gdnative-derive/src/derive_macro.rs

```diff
@@ -48,13 +60,50 @@ pub(crate) fn parse_derive_input(input: TokenStream) -> DeriveData {
     };
 
     // read exported properties
-    if let Fields::Named(_names) = struct_data.fields {
-        // TODO
-    }
+    let props = if let Fields::Named(names) = &struct_data.fields {
+        names
+            .named
+            .iter()
+            .map(|field| (
+                field.clone(),
+                field.attrs.iter().filter(|attribute| attribute.path.is_ident("property"))
+                    .map(|x| x.clone())
+                    .collect::<Vec<_>>()
+            ))
+            .filter(|(_, attributes)| attributes.len() > 0)
+            .map(|(field, attributes)| {
+                let mut map = HashMap::new();
+                for attribute in attributes.iter() {
+                    attribute.parse_meta().ok()
+                        .map(|meta| if let syn::Meta::List(c) = meta { Some(c.nested.clone()) } else { None })
+                        .filter(|meta| meta.is_some())
+                        .map(|meta| {
+                            for entry in meta.unwrap().iter() {
+                                match &entry {
+                                    &syn::NestedMeta::Meta(syn::Meta::NameValue(pair)) => {
+                                        let ident = pair.path.get_ident();
+                                        if ident.is_some() {
+                                            map.insert(ident.unwrap().to_string(), pair.lit.clone());
+                                        }
+                                    },
+                                    _ => {}
+                                }
+                            }
+                        })
+                        .expect("property recieved no parameters...")
+                }
+                (field, map)
+            })
+            .collect::<HashMap<_,_>>()
+    } else {
+        HashMap::new()
+    };
```

Comment:
This whole `map` `clone` `collect` `filter` `map` sequence with a lot of allocation is unnecessary. Silent failures are also bad: we might move to something like [`proc-macro-error`](https://crates.io/crates/proc-macro-error) later, but you should at least panic in case of error in a procedural macro. There is `Iterator::filter_map` with which you can do:

```rust
// Compiles and passes tests with some minor changes elsewhere. You can copy these.

names
    .named
    .iter()
    .filter_map(|field| {
        let mut property_args = None;

        for attr in field.attrs.iter() {
            if !attr.path.is_ident("property") {
                continue;
            }

            let meta = attr.parse_meta().expect("should be able to parse attribute arguments");
            if let Meta::List(MetaList { nested, .. }) = meta {
                property_args
                    .get_or_insert_with(PropertyAttrArgsBuilder::default)
                    .extend(nested.iter().map(|arg| {
                        match arg {
                            NestedMeta::Meta(Meta::NameValue(ref pair)) => pair,
                            _ => panic!("unexpected argument: {:?}", arg),
                        }
                    }));
            } else {
                panic!("unexpected meta variant: {:?}", meta);
            }
        }

        property_args.map(|builder| {
            let ident = field.ident.clone().expect("fields should be named");
            (ident, builder.done()) // Avoid cloning the whole `Field` since only the identifier is needed in implementation.
        })
    })
    .collect::<HashMap<_, _>>()
```

Where `PropertyAttrArgs` (see other comment) and `PropertyAttrArgsBuilder` are implemented like:

```rust
// Compiles and passes tests with some minor changes elsewhere. You can copy these.

#[derive(Default)]
struct PropertyAttrArgsBuilder {
    default: Option<Lit>,
}

pub(crate) struct PropertyAttrArgs {
    pub default: Lit,
}

impl<'a> Extend<&'a MetaNameValue> for PropertyAttrArgsBuilder {
    fn extend<I>(&mut self, iter: I)
    where
        I: IntoIterator<Item = &'a MetaNameValue>,
    {
        for pair in iter.into_iter() {
            let name = pair.path.get_ident().expect("should be single identifier").to_string();
            match name.as_str() {
                "default" => {
                    if let Some(old) = self.default.replace(pair.lit.clone()) {
                        panic!("there is already a default value set: {:?}", old);
                    }
                },
                _ => panic!("unexpected argument: {}", &name),
            }
        }
    }
}

impl PropertyAttrArgsBuilder {
    fn done(self) -> PropertyAttrArgs {
        PropertyAttrArgs {
            default: self.default.expect("`default` value is required"),
        }
    }
}

```